app/helpers/training_helper.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+#= Helpers for training views
+module TrainingHelper
+  # Given a module slug, it returns the first library that has a
+  # category including that module slug. It returns nil if no such
+  # library is found.
+  def find_library_from_module_slug(module_slug)
+    TrainingLibrary.all.find_each do |library|
```

Comment:
For this method, I would suggest implementing TrainingLibrary#training_module_slugs (for example:
```
  def training_module_slugs
    categories.map do |cat|
      cat['modules'].map { |mod| mod['slug'] }
    end.flatten
  end
```
)

and then using that to implement TrainingModule#library:

```
TrainingLibrary.all.detect { |tl| tl.training_module_slugs.include? slug }
```